src/main/java/org/folio/mr/service/impl/StaffSlipContextService.java

```diff
@@ -0,0 +1,110 @@
+package org.folio.mr.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.mr.domain.dto.Instance;
+import org.folio.mr.domain.dto.InstanceContributorsInner;
+import org.folio.mr.domain.dto.Item;
+import org.folio.mr.domain.dto.MediatedRequest;
+import org.folio.mr.domain.dto.SendItemInTransitResponseStaffSlipContext;
+import org.folio.mr.domain.dto.SendItemInTransitResponseStaffSlipContextItem;
+import org.folio.mr.service.InventoryService;
+import org.springframework.stereotype.Service;
+
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class StaffSlipContextService {
+
+  private final InventoryService inventoryService;
+
+  public SendItemInTransitResponseStaffSlipContext createStaffSlipContext(MediatedRequest request) {
+    Item item = inventoryService.fetchItem(request.getItemId());
+
+    log.debug("createStaffSlipContext:: parameters item: {}", item);
+
+    var holding = inventoryService.fetchHolding(item.getHoldingsRecordId());
+    var instance = inventoryService.fetchInstance(holding.getInstanceId());
+    var materialType = inventoryService.fetchMaterialType(item.getMaterialTypeId());
+    var loanType = inventoryService.fetchLoanType(item.getPermanentLoanTypeId());
+    var toServicePoint = inventoryService.fetchServicePoint(item.getInTransitDestinationServicePointId());
```

Comment:
Can any of these be null (including item)? If they can, could this cause an NPE?

Reply:
What I found by looking into docs an tested with IT (MediatedRequestActionsApiTest) FeignException.NotFound will be thrown but not handled by ApiExceptionHandler. I will add FeignException handler:
![image](https://github.com/user-attachments/assets/dfd47f35-f98a-43fa-8b1f-6cdf82b8a0c3)